src/Amadevus.RecordGenerator.Generators/GeneratedCodeAttributeGenerator.cs

```diff
@@ -119,21 +119,9 @@ private AttributeArgumentListSyntax GenerateAttributeArgumentList()
                                 Literal(toolVersion)))}));
         }
 
-        private QualifiedNameSyntax GenerateQualifiedName()
+        private NameSyntax GenerateQualifiedName()
         {
-            var namespaceAndIdentifier = Names.GeneratedCodeAttribute.Split('.')
-                .Select(n => IdentifierName(n)).ToList();
-
-            var attributeName = QualifiedName(
-                namespaceAndIdentifier.First(),
-                namespaceAndIdentifier.Skip(1).First());
-
-            foreach (var identifier in namespaceAndIdentifier.Skip(2))
-            {
-                attributeName = QualifiedName(attributeName, identifier);
-            }
-
-            return attributeName; 
+            return QualifiedNameGenerator.GenerateQualifiedName(Names.GeneratedCodeAttribute);
```

Comment:
I think you can just do [`ParseName`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntaxfactory.parsename?view=roslyn-dotnet) instead of creating a whole custom class.

Reply:
Didn't know of that one. Used ist whereever possible